pkg/controllers/provisioning/v1alpha1/reallocation/terminate.go

```diff
@@ -30,39 +30,50 @@ type Terminator struct {
 	cloudprovider cloudprovider.Factory
 }
 
-func (t *Terminator) CordonNodes(ctx context.Context, nodes []*v1.Node) error {
+// AddTTLs takes in a list of underutilized nodes and adds TTL to them
+func (t *Terminator) AddTTLs(ctx context.Context, nodes []*v1.Node) error {
+	if len(nodes) == 0 {
+		return nil
+	}
 	for _, node := range nodes {
-		if node.Spec.Unschedulable {
-			continue
-		}
-		persisted := node.DeepCopyObject()
-		node.Spec.Unschedulable = true
+		persisted := node.DeepCopy()
+		node.Annotations[v1alpha1.ProvisionerTTLKey] = time.Now().Add(300 * time.Second).Format(time.RFC3339)
 		if err := t.kubeClient.Patch(ctx, node, client.MergeFrom(persisted)); err != nil {
-			return fmt.Errorf("patching node %s, %w", node.Name, err)
+			return fmt.Errorf("patching node, %w", err)
 		}
 	}
+	zap.S().Debugf("Added TTL to %d nodes", len(nodes))
 	return nil
 }
 
-func (t *Terminator) AddTTLs(ctx context.Context, nodes []*v1.Node) error {
+// CordonNodes takes in a list of expired nodes as input and cordons them
+func (t *Terminator) CordonNodes(ctx context.Context, nodes []*v1.Node) error {
+	if len(nodes) == 0 {
+		return nil
+	}
 	for _, node := range nodes {
-		if _, ok := node.Annotations[v1alpha1.ProvisionerTTLKey]; !ok {
-			persisted := node.DeepCopy()
-			node.Annotations[v1alpha1.ProvisionerTTLKey] = time.Now().Add(300 * time.Second).Format(time.RFC3339)
-			if err := t.kubeClient.Patch(ctx, node, client.MergeFrom(persisted)); err != nil {
-				return fmt.Errorf("patching node, %w", err)
-			}
+		persisted := node.DeepCopy()
+		node.Spec.Unschedulable = true
+		if err := t.kubeClient.Patch(ctx, node, client.MergeFrom(persisted)); err != nil {
+			return fmt.Errorf("patching node %s, %w", node.Name, err)
```

Comment:
"patching" -> "cordoning"? (More closely matches "kubectl cordon")

Reply:
the level above is logging "cordoning node", so switching this to cordoning would be redundant. To return something unique, I think patching makes sense.

Reply:
I think it's going to be confusing (I'm just thinking thru the looking in logs at 3am case during an outage..) - eg at a glance at this log line how to distinguish it from the patching node on line 39 (for ttl)?

Reply:
Since the error message has the `%w` in it, it'll be appended to the other error messages that are being returned in the stack trace. So you would see `cordoning node, patching node, err` rather than just solely `patching node, error`.

Reply:
Ah ok